table.go

```diff
@@ -264,12 +264,15 @@ func addPointsField(fields core.Fields) core.Fields {
 }
 
 func (t *table) startFollowing(walOffset wal.Offset) {
+	t.db.followMx.Lock()
 	newSubscriber := t.db.newStreamSubscriber[t.From]
 	if newSubscriber == nil {
 		newSubscriber = make(chan *tableWithOffset, 100)
+		log.Debugf("Following leader for %v", t.From)
 		go t.db.followLeader(t.From, newSubscriber)
```

Comment:
This is the actual fix. It used to be possible for us to spawn multiple `followLeader` calls for the same stream. The mutex prevents that.